unimport/files.py

```diff
@@ -1,13 +1,25 @@
 import os
+import re
 
-from unimport.config import is_ignore_files, is_ignore_folder
 
+def get_files(src, config):
+    ignored_folders = re.compile("|".join(set(config.ignored_folders)))
+    ignored_files = re.compile("|".join(set(config.ignored_files)))
 
-def get_files(direction):
-    for root, dirs, files in os.walk(direction):
-        if is_ignore_folder(root):
+    def _is_ignored_folder(path):
+        return ignored_folders.match(path) is not None
+
+    def _is_ignored_file(path):
+        return not path.endswith(".py") or ignored_files.match(path) is not None
+
+    if not src.is_dir():
+        return str(src) if not _is_ignored_file(src.name) else None
+
+    for root, dirs, files in os.walk(str(src.absolute())):
```

Comment:
It is really unpractical to mix `os.path` with `pathlib`. As far as I am aware @hakancelik96 used `os.path` in the other places at codebase so if you are planning to use `pathlib` it would be better to refactor all (I am in favor of pathlib because it is way better than `os.path` in terms of readability but please dont mix them.)

Reply:
@isidentical I agree with you, now I created a separate issue for that topic to keep this PR simple. Let's go ahead step by step.

https://github.com/hakancelik96/unimport/issues/8